File: python/nth_prime.py
import inout as io

def prime(n):
    primes = [2]
    suspected = 3

    while len(primes) < n:
        sus_is_prime = True
        for i in primes[1:]:
            if suspected % i == 0:
                sus_is_prime = False
                break
        
        if sus_is_prime:
            primes.append(suspected)
        suspected += 2

    return primes[-1]


def write(n, ans):
    if n%10==1:
        print(f'{n}-sza liczba pierwsza: {ans}')
    elif n%10==2:
        print(f'{n}-ga liczba pierwsza: {ans}')
    elif n%10==3:
        print(f'{n}-cia liczba pierwsza: {ans}')
    elif n%100==0:
        print(f'{n}-na liczba pierwsza: {ans}')
    else:
        print(f'{n}-sza liczba pierwsza: {ans}')


def main():
    n, = io.get_input(int)
    ans = prime(n)

    write(n, ans)

    
if __name__ == '__main__':
    main()

File: c/src/nth_prime.c
#include <stdio.h>
#include <stdbool.h>

bool input(unsigned *n, int argc, char const *argv[]) {
    switch (argc) {
        case 1:
            printf("Podaj liczbę: ");
            if (scanf("%u", n)!=1 || *n<=0) {
                printf("Niepoprawna forma argumentu\n");
                return false;
            };
            break;

        case 2:
            if (sscanf(argv[1], "%u", n)!=1 || *n<=0) {
                printf("Niepoprawna forma argumentu\n");
                return false;
            }
            break; 
        
        default:
            printf("Niepoprawna liczba argumentów\n");
            return false;
    }
    return true;
}

void output(unsigned n, unsigned primes) {
    switch (n % 10) {
        case 1:
            printf("%d-sza liczba pierwsza: %d\n", n, primes);
            break;
        
        case 2:
            printf("%d-ga liczba pierwsza: %d\n", n, primes);
            break;

        case 3:
            printf("%d-cia liczba pierwsza: %d\n", n, primes);
            break;

        default:
            if (n%10 == 0) {
                printf("%d-na liczba pierwsza: %d\n", n, primes);
                break;
            }
            printf("%d-ta liczba pierwsza: %d\n", n, primes);
            break;
    }
}

unsigned nth_prime(unsigned m) {
    unsigned n = m;
    unsigned primes[n], suspected = 3, counter = 1;
    primes[0] = 2;
    bool sus_is_prime;

    while (counter < n) {
        sus_is_prime = true;
        for (int i = 1; i < counter; i++) {
            if (suspected % primes[i] == 0) {
                sus_is_prime = false;
                break;
            }
        }

        if (sus_is_prime) {
            primes[counter] = suspected;
            counter++;
        }
        suspected += 2;
    }

    return primes[counter-1];
}

int main(int argc, char const *argv[])
{
    unsigned n;
    if (!input(&n, argc, argv)) { return -1; }

    unsigned prime = nth_prime(n);

    output(n, prime);
    return 0;
}
